### ISSUE TITLE
The method expects does only exist in PHPUnit_Framework_MockObject_MockObject, but not in Acme\Foo.

### ISSUE THREAD

**Issue by Maks3w** [scrutinizer-ci/php-analyzer#653]
```php
   /**
    * @var Acme\Foo|\PHPUnit_Framework_MockObject_MockObject
    */
    protected $foo;

    function testFoo() {
        // The method expects does only exist in PHPUnit_Framework_MockObject_MockObject, but not in Acme\Foo.
        $this->foo->expects($this->once())->method('find');
        $this->foo->find();
    }
```

This kind of duble type a PHPUnit mock shouldn't throw any warning.
